### ISSUE TITLE
Embedded content not displaying

### ISSUE THREAD

**Issue by powdermonkeydan** [brandaorodrigo/moodle-format_buttons#14]
Hi!

Love the Buttons course format. We are working with H5P quite a lot and finding a couple of issues:

1. The window needs to be resized to show the H5P activities (we read this might be the resize script, which we have removed, but no improvement)
2. When clicking on a button to go to a different section, the H5P content doesn't load. We just see a blank screen. User has to reload to view it.

Video attached as zip.

[buttons issue.mp4.zip](https://github.com/brandaorodrigo/moodle-format_buttons/files/1548460/buttons.issue.mp4.zip)

Is this a bug or are we doing something wrong? Thanks.

**Comment by brandaorodrigo** [comment]
Hello, the project was frozen a few months. But now I'm returning. I'm wondering if you can create a student user to me to access your course and try to figure out the issue. You can send me a private message.

**Comment by powdermonkeydan** [comment]
Hi Rodrigo

Happy to do this for you. Can you give me an email address and I'll create a course and send you account login details. Thanks.

**Comment by powdermonkeydan** [comment]
Rodrigo - go to http://apps.improveinternational.com/moodle/course/view.php?id=25 - I have setup Guest access, so you can login via Guest and only view this course. Hope that helps.

**Comment by powdermonkeydan** [comment]
Sorry - forgot to mention a couple of things:

1. This is the embed code for each H5P label: `<iframe src="https://h5p.5mcreative.com/wp-admin/admin-ajax.php?action=h5p_embed&amp;id=32" width="914" height="539" frameborder="0" allowfullscreen="allowfullscreen"></iframe><script src="https://h5p.5mcreative.com/wp-content/plugins/h5p/h5p-php-library/js/h5p-resizer.js" charset="UTF-8"></script>`
2. If you resize the window, the missing H5P lesson appears
3. This seems to work fine in mobile mode (it looks like each 'page' refreshes itself, which doesn't happen on the web version).

**Comment by brandaorodrigo** [comment]
Thanks!!! 

I'm going to replicate your iframe code in my localhost and I'll work around that problem.

**Comment by brandaorodrigo** [comment]
Hello! I've released a new version (Build: 2018022400) in Moodle.org and GitHub. In this version, I'm added support to H5P contents and fix others tiny bugs. Please, try it and send to me the result. Thanks again!

**Comment by powdermonkeydan** [comment]
Hi!


Thanks for this. I’ve just tested and it’s still the same unfortunately - I have to resize the window slightly to show the H5P content.


Could it be theme related? We use Lambda (purchased theme but very stable).


Thanks - Dan


On 25 February 2018 at 06:37:41, Rodrigo Brandão (notifications@github.com<mailto:notifications@github.com>) wrote:

Hello! I've released a new version (Build: 2018022400) in Moodle.org and GitHub. In this version, I'm added support to H5P contents and fix others tiny bugs. Please, try it and send to me the result. Thanks again!

**Comment by brandaorodrigo** [comment]
Maybe, but is difficult to be a theme problem. I’ll login on guest mode again to see the issue.

**Comment by powdermonkeydan** [comment]
Thanks.


I have actually tested it on the site I gave you access to, I updated the plugin on another site (3.3) where I have courses with H5P embedded.


Let me test it on the site I gave you access to as well :)


On 25 February 2018 at 15:38:01, Rodrigo Brandão (notifications@github.com<mailto:notifications@github.com>) wrote:

Maybe, but is difficult to be a theme problem. I’ll login on guest mode again to see the issue.

**Comment by powdermonkeydan** [comment]
OK just tested it on 3.2 and that DOES seem to be working.


Just not on 3.3 :(


On 25 February 2018 at 15:43:01, Dan Jeffries (dan.jeffries@improveinternational.com<mailto:dan.jeffries@improveinternational.com>) wrote:

Thanks.


I have actually tested it on the site I gave you access to, I updated the plugin on another site (3.3) where I have courses with H5P embedded.


Let me test it on the site I gave you access to as well :)


On 25 February 2018 at 15:38:01, Rodrigo Brandão (notifications@github.com<mailto:notifications@github.com>) wrote:

Maybe, but is difficult to be a theme problem. I’ll login on guest mode again to see the issue.

**Comment by brandaorodrigo** [comment]
Apparently, it works on the site where I can log in to the guest mode.

Can you give me the URL of this another installation where working?

I figure out the problem with the H5P - The javascript who resizes the iframe runs one time when the loading of the page ends and just find and resize the visible iframes. What I did was request a new load of the H5p script when the user clicks on every section button.

**Comment by brandaorodrigo** [comment]
* The h5p scripts run too when the browser changes the window size.

**Comment by powdermonkeydan** [comment]
Ah OK - understood.


So if you go to:


https://moodleuk.improveinternational.com/course/view.php?id=164


Login: test2

password: Pa55w0rd!


I remember someone saying about putting the H5P JS in the site header? Would that help? (I’m not very good with JS).


Thanks for the help with this - appreciate it!


On 25 February 2018 at 16:03:03, Rodrigo Brandão (notifications@github.com<mailto:notifications@github.com>) wrote:

Apparently, it works on the site where I can log in to the guest mode.

Can you give me the URL of this another installation where working?

I figure out the problem with the H5P - The javascript who resizes the iframe runs one time when the loading of the page ends and just find and resize the visible iframes. What I did was request a new load of the H5p script when the user clicks on every section button.

**Comment by brandaorodrigo** [comment]
Hello Dan, I found the problem: Your Moodle is using old files of my plugin.
Look:https://moodleuk.improveinternational.com/lib/javascript.php/1519569754/course/format/buttons/module.js

I have two ideas: 

(1) The cache of your installation doesn't refresh all javascript files. - Please go o Administrator > Developer > Purge all cache. And test again.

(2) If idea 1 not works: Do a full uninstall of my plugin and look if the files in  /course/format/buttons were deleted. If not, delete it and reinstall my plugin.

**Comment by powdermonkeydan** [comment]
Ah great - thanks.


We actually use a multi tenant system where one site updates all other plugins. I’ll pass this on to the IT team who will hopefully resolve it.


I’ll get back to you :)


On 25 February 2018 at 20:12:24, Rodrigo Brandão (notifications@github.com<mailto:notifications@github.com>) wrote:

Hello Dan, I found the problem: Your Moodle is using old files of my plugin.
Look:https://moodleuk.improveinternational.com/lib/javascript.php/1519569754/course/format/buttons/module.js

I have two ideas:

(1) The cache of your installation doesn't refresh all javascript files. - Please go o Administrator > Developer > Purge all cache. And test again.

(2) If idea 1 not works: Do a full uninstall of my plugin and look if the files in /course/format/buttons were deleted. If not, delete it and reinstall my plugin.

**Comment by powdermonkeydan** [comment]
Hi!


We have now updated the plugin on all our sites and H5P is now showing in every section.


THANK YOU! It’s a big fix for us.


Keep up the good work :)


Dan


On 25 February 2018 at 20:20:30, Dan Jeffries (dan.jeffries@improveinternational.com<mailto:dan.jeffries@improveinternational.com>) wrote:

Ah great - thanks.


We actually use a multi tenant system where one site updates all other plugins. I’ll pass this on to the IT team who will hopefully resolve it.


I’ll get back to you :)


On 25 February 2018 at 20:12:24, Rodrigo Brandão (notifications@github.com<mailto:notifications@github.com>) wrote:

Hello Dan, I found the problem: Your Moodle is using old files of my plugin.
Look:https://moodleuk.improveinternational.com/lib/javascript.php/1519569754/course/format/buttons/module.js

I have two ideas:

(1) The cache of your installation doesn't refresh all javascript files. - Please go o Administrator > Developer > Purge all cache. And test again.

(2) If idea 1 not works: Do a full uninstall of my plugin and look if the files in /course/format/buttons were deleted. If not, delete it and reinstall my plugin.

**Comment by brandaorodrigo** [comment]
\o/

Thank to use my plugin.
